### PR TITLE
RTL3a

### PR DESCRIPTION
**Pull request by ricardopereira**
❌ Not passing.

### CODE REVIEW

File: ablySpec/RealtimeClientChannel.swift [hunk 1]
```diff
@@ -13,6 +13,63 @@ class RealtimeClientChannel: QuickSpec {
...
+
+                        let channel = client.channel("test")
+                        channel.attach()
+                        expect(channel.state).toEventually(equal(ARTRealtimeChannelState.Attached), timeout: testTimeout)
+
+                        waitUntil(timeout: testTimeout) { done in
+                            channel.subscribeToStateChanges { state, status in
+                                if state == .Failed {
+                                    expect(status.state).to(equal(ARTState.Error))
+                                    expect(status.errorInfo!.code).to(equal(90000))
```
Comment (mattheworiordan) [inline ablySpec/RealtimeClientChannel.swift]: Might be nice in both of these tests to simply double check that the channel state itself as failed as opposed to simply emitting the failed state.  I appreciate this is probably covered elsewhere, but the test explicitly states that the channel should transition to failed, and as such, it's nice to check that it both emits a failed state and has a failed state.

**Comment by mattheworiordan** [comment]
LGTM other than one comment

File: ablySpec/RealtimeClientChannel.swift (hunk 1 above)
Comment (ricardopereira) [inline ablySpec/RealtimeClientChannel.swift]: Done

Comment (tcard) [inline ablySpec/RealtimeClientChannel.swift]: > Done

Hmm, did you forget to push maybe? It's still the same.

**Comment by tcard** [comment]
LGTM

File: ablySpec/RealtimeClientChannel.swift (hunk 1 above)
Comment (ricardopereira) [inline ablySpec/RealtimeClientChannel.swift]: I edit the commit: https://github.com/ably/ably-ios/commit/8352fd528335672cc2a122805016e94e6518e380#diff-74dac661e35645777f94b2a135363313R68

**Pull request merged**